agency/detail/tuple_impl.hpp

```diff
@@ -160,13 +160,13 @@ template<class T, bool = tuple_use_empty_base_class_optimization<T>::value>
 class tuple_leaf_base
 {
   public:
-#if defined(__CUDACC__)
+#if defined(__CUDACC__) && !(defined(__clang__) && defined(__CUDA__))
```

Comment:
I agree. I don't know how to isolate nvcc code path, as clang also defines `__NVCC__`. As with  Thrust,  we  it the following way

    #if defined(__NVCC__) && !(defined(__clang__) && defined(__CUDA__))

If that's okay with you, I can fix PR.